### ISSUE TITLE
Fix for eslint error "Do not use setState in componentDidMount       react/no-did-mount-set-state"

### ISSUE THREAD

**Issue by stevus** [tomchentw/react-google-maps#141]
In [L81 (and other map/components})](https://github.com/tomchentw/react-google-maps/blob/master/src/GoogleMap.js#L81):

should be instead:
```
componentWillReceiveProps (nextProps) {

    if(typeof this.state.map === 'undefined') {
        const domEl = findDOMNode(this);
        const {containerTagName, containerProps, children, ...mapProps} = nextProps;
        const map = GoogleMapHolder._createMap(domEl, mapProps);
        this.setState({ map });
    }
}
```

otherwise will fail ES2015 eslinting

```
error  Do not use setState in componentDidMount       react/no-did-mount-set-state
```

**Comment by tomchentw** [comment]
We really want it to happen in `cDM` because we need DOM node for google maps instance. Any ideas?

**Comment by tomchentw** [comment]
@stevus How about using callback style ref? I've opened up a branch for this: https://github.com/tomchentw/react-google-maps/compare/fix-141?expand=1

**Comment by tomchentw** [comment]
Use callback style ref in the new refactoring branch #157 .

**Comment by tomchentw** [comment]
Released [v4.6.0](https://github.com/tomchentw/react-google-maps/releases/tag/v4.6.0)

**Comment by stevus** [comment]
Awesome work, and quick response. I will try it out.
